File: src/code-samples/langsmith/smithdb-migration/runs-query-fetch-by-id-before.go
// :snippet-start: runs-query-fetch-by-id-before-go
// :codegroup-tab: Before
package main

import (
	"context"

	"github.com/langchain-ai/langsmith-go"
)

// :remove-start:
func main() {
// :remove-end:
ctx := context.Background()
client := langsmith.NewClient()

sessions, err := client.Sessions.List(ctx, langsmith.SessionListParams{
	Name:  langsmith.F("default"),
	Limit: langsmith.F(int64(1)),
})
// :remove-start:
if err != nil {
	panic(err.Error())
}
// :remove-end:
project := sessions.Items[0]

runID1 := "<run-id-1>"
runID2 := "<run-id-2>"
// :remove-start:
found, err := client.Runs.Query(ctx, langsmith.RunQueryParams{
	Session: langsmith.F([]string{project.ID}),
	Limit:   langsmith.F(int64(2)),
})
if err != nil {
	panic(err.Error())
}
runID1 = found.Runs[0].ID
runID2 = found.Runs[1].ID
// :remove-end:
runs, err := client.Runs.Query(ctx, langsmith.RunQueryParams{
	Session: langsmith.F([]string{project.ID}),
	ID:      langsmith.F([]string{runID1, runID2}),
})
// :remove-start:
if err != nil {
	panic(err.Error())
}
_ = runs
// :remove-end:
// :remove-start:
}

// :remove-end:
// :snippet-end:

File: src/code-samples/langsmith/smithdb-migration/runs-query-fetch-by-id-before.kt
///usr/bin/env jbang "$0" "$@" ; exit $?
//JAVA 21
//KOTLIN 2.2.0
//DEPS com.langchain.smith:langsmith-java:0.1.0-beta.11

// :snippet-start: runs-query-fetch-by-id-before-kt
// :codegroup-tab: Before
import com.langchain.smith.client.LangsmithClient
import com.langchain.smith.client.okhttp.LangsmithOkHttpClient
import com.langchain.smith.models.runs.RunQueryParams
import com.langchain.smith.models.sessions.SessionListParams

// :remove-start:
fun main() {
    if (System.getenv("LANGSMITH_API_KEY").isNullOrBlank()) {
        println("[smithdb-runs-query-fetch-by-id-before] Skipping (LANGSMITH_API_KEY is not set).")
        return
    }

// :remove-end:
val client: LangsmithClient = LangsmithOkHttpClient.fromEnv()

val project = client.sessions().list(
    SessionListParams.builder().name("default").limit(1L).build()
).items().first()
var runId1 = "<run-id-1>"
var runId2 = "<run-id-2>"
// :remove-start:
val foundRuns = client.runs().query(
    RunQueryParams.builder().addSession(project.id()).limit(2L).build()
).items()
runId1 = foundRuns[0].id()
runId2 = foundRuns[1].id()
// :remove-end:
val runs = client.runs().query(
    RunQueryParams.builder()
        .addSession(project.id())
        .addId(runId1)
        .addId(runId2)
        .build()
).items()
// :remove-start:
}
// :remove-end:
// :snippet-end:
